core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
A new data structure with floats just works with Go. A new data structure with a float in GDScript on the other hand, if will just crash Godot unless they repeat the behavior you expect using a custom Comparator.

So it seems only natural to me that you need to document this pitfall.